src/image.cpp

```diff
@@ -84,6 +84,55 @@ void image::resize(const cv::Mat& input, cv::Mat& output, const cv::Size2i& size
     }
 }
 
+void image::resize_short(const cv::Mat& input, cv::Mat& output, const int target_size)
+{
+    if (target_size == 0)
+    {
+        output = input;
+        return;
+    }
+
+    auto percent = static_cast<float>(target_size) / std::min(input.cols, input.rows);
+
+    auto resized_width  = static_cast<int>(std::round(input.cols * percent));
+    auto resized_height = static_cast<int>(std::round(input.rows * percent));
+    cv::resize(input, output, cv::Size2i(resized_width, resized_height), 0, 0, CV_INTER_LINEAR);
+}
+
+void image::standardize(std::vector<cv::Mat>&      input,
+                        const std::vector<double>& mean,
+                        const std::vector<double>& stddev)
+{
+    if (mean.empty())
+        return;
+    // prep for division
+    cv::Scalar s_mean, s_stddev;
+    for (int i = 0; i < mean.size(); i++)
+    {
+        s_stddev[i] = stddev[i] ? 1. / stddev[i] : 1.;
+        s_mean[i]   = mean[i];
+    }
+
+    for (int i = 0; i < input.size(); i++)
+    {
+        // divide
+        multiply(input[i], 1. / 255., input[i]);
+
+        if (input[i].channels() == 1)
+        {
+            subtract(input[i], s_mean[i], input[i]);
+            // divide
+            multiply(input[i], s_stddev[i], input[i]);
+        }
+        else
+        {
+            subtract(input[i], s_mean, input[i]);
+            // divide
+            multiply(input[i], s_stddev, input[i]);
+        }
```

Comment:
same problem hear as below, 4th element of s_mean is undefined. and how could OpenCV  know that you have 3th element vector ??

Reply:
I check that if mean and stddev are used then the length of each must be equal to number of channels in https://github.com/NervanaSystems/aeon/pull/6/files#diff-a4ccfbf4e57f797ce3bbb8864ceaf6cbR223